### PR TITLE
Eliminate "scope" in ApplicationGenerator

### PR DESCRIPTION
**Pull request by benjaminapetersen**
"scope" was the object returned from the ApplicationGenerator service.  This is a misnomer & is rather confusing. Updated to return an object with "pubic" methods, leaving all underscored methods private.

@jwforres @spadgett for review, noticed while doing key-value-editor replacements.

### CODE REVIEW

**Comment by spadgett** [comment]
you'll need to update applicationGeneratorSpec.js

File: dist/scripts/templates.js [hunk 1]
```diff
@@ -3613,6 +3613,7 @@ angular.module('openshiftConsoleTemplates', []).run(['$templateCache', function(
     "<i class=\"pficon pficon-help\"></i>\n" +
     "</a>\n" +
     "</span></h3>\n" +
+    "{{buildConfig.envVars | json}}\n" +
```
Comment (spadgett) [inline dist/scripts/templates.js]: Looks like an unrelated change in dist?

Comment (benjaminapetersen) [inline dist/scripts/templates.js]: ah, bother grunt build :) will fix.

**Comment by benjaminapetersen** [comment]
@spadgett udpated

**Comment by spadgett** [comment]
Still needs applicationGeneratorSpec.js test updates

**Comment by spadgett** [comment]
@benjaminapetersen ping :)

**Comment by benjaminapetersen** [comment]
Thanks @spadgett, will refresh this one quick!

**Comment by benjaminapetersen** [comment]
@spadgett most test easy to fix, but this one fails:
`ApplicationGenerator generating applications from image that includes source should generate a BuildConfig for the source`
with the difference being some generated in triggers:
```javascript
triggers: [{
            generic: {
                secret: 'c77e0419e36c8478'
            },
            type: 'Generic'
        }, {
            github: {
                secret: '26e64b71938ea292'
            },
            type: 'GitHub'
        }, {
            imageChange: {},
            type: 'ImageChange'
        }, {
            type: 'ConfigChange'
        }]
```
vs
```javascript
triggers: [{
            generic: {
                secret: 'secret101'
            },
            type: 'Generic'
        }, {
            github: {
                secret: 'secret101'
            },
            type: 'GitHub'
        }, {
            imageChange: {},
            type: 'ImageChange'
        }, {
            type: 'ConfigChange'
        }]
```
Working through this. I dont know atm why my change would affect this.

**Comment by benjaminapetersen** [comment]
Ah, wait, maybe i do.

**Comment by benjaminapetersen** [comment]
Ok @spadgett this is working again.  Had to think a bit about the monkey patch of `ApplicationGenerator.generateSecret()` after the change. Internally its called as a standalone fn (`generateSecret()`), so replacing the whole method in the test file was no longer working.

File: app/scripts/services/applicationGenerator.js [hunk 2]
```diff
@@ -395,13 +349,71 @@ angular.module("openshiftConsole")
...
+      return resources;
+    };
+
+    return {
+      createRoute: createRoute,
+      generate: generate,
+      parsePorts: parsePorts,
+      generateSecret: generateSecret,
+      getServicePort: getServicePort,
+      // pseudo private
```
Comment (spadgett) [inline app/scripts/services/applicationGenerator.js]: pseudo private == needed for tests? If so, I'd say that in the comment.

File: test/spec/services/applicationGeneratorSpec.js [hunk 3]
```diff
@@ -3,13 +3,23 @@
 describe("ApplicationGenerator", function(){
   var ApplicationGenerator;
   var inputTemplate;
+  var mathFloor = Math.floor;
 
   beforeEach(function(){
 
     inject(function(_ApplicationGenerator_){
       ApplicationGenerator = _ApplicationGenerator_;
-      ApplicationGenerator._generateSecret = function(){
```
Comment (spadgett) [inline test/spec/services/applicationGeneratorSpec.js]: Is there a reason we can't simply change this to `ApplicationGenerator.generateSecret` and leave the mock as before?

File: app/scripts/services/applicationGenerator.js (hunk 2 above)
Comment (benjaminapetersen) [inline app/scripts/services/applicationGenerator.js]: right... we have tests already written so I didn't want to just remove the tests.  Perhaps they dont even need to be semi-private?

File: test/spec/services/applicationGeneratorSpec.js (hunk 3 above)
Comment (benjaminapetersen) [inline test/spec/services/applicationGeneratorSpec.js]: Yup. Since the end of the file does this:

```javascript
return {
  // "exporting" the functions...
  _generateSecret: generateSecret
}
```
Doing `ApplicationGenerator._generateSecret = function(){ ... }` is not effective.  The original function is used internally a number of times: 

Example:
```javascript 
// line 233, this call cannot be mocked
secret: generateSecret()  
``` 
So will never use the mocked function.

File: app/scripts/services/applicationGenerator.js (hunk 2 above)
Comment (benjaminapetersen) [inline app/scripts/services/applicationGenerator.js]: updated comment

File: test/spec/services/applicationGeneratorSpec.js (hunk 3 above)
Comment (benjaminapetersen) [inline test/spec/services/applicationGeneratorSpec.js]: @spadgett more thoughts on this?

**Comment by benjaminapetersen** [comment]
Hmm... gonna close this. The rebase looks like a train wreck, prob easier to just redo.
